src/server/game/Maps/TransportMgr.cpp

```diff
@@ -404,51 +404,57 @@ void TransportMgr::SpawnContinentTransports()
         return;
 
     uint32 oldMSTime = getMSTime();
-    QueryResult result = WorldDatabase.Query("SELECT guid, entry FROM transports");
     uint32 count = 0;
-    if (result)
+    if (sWorld->getBoolConfig(CONFIG_ENABLE_CONTINENT_TRANSPORT))
     {
-        do
+        QueryResult result = WorldDatabase.Query("SELECT guid, entry FROM transports");
+        if (result)
         {
-            Field* fields = result->Fetch();
-            uint32 guid = fields[0].GetUInt32();
-            uint32 entry = fields[1].GetUInt32();
-
-            if (TransportTemplate const* tInfo = GetTransportTemplate(entry))
-                if (!tInfo->inInstance)
-                    if (CreateTransport(entry, guid))
-                        ++count;
+            do
+            {
+                Field* fields = result->Fetch();
+                uint32 guid = fields[0].GetUInt32();
+                uint32 entry = fields[1].GetUInt32();
 
-        } while (result->NextRow());
-    }
+                if (TransportTemplate const* tInfo = GetTransportTemplate(entry))
+                    if (!tInfo->inInstance)
+                        if (CreateTransport(entry, guid))
+                            ++count;
 
-    sLog->outString(">> Spawned %u continent motion transports in %u ms", count, GetMSTimeDiffToNow(oldMSTime));
+            } while (result->NextRow());
+        }
 
-    // pussywizard: preload grids for continent static transports
-    oldMSTime = getMSTime();
-    result = WorldDatabase.Query("SELECT map, position_x, position_y FROM gameobject g JOIN gameobject_template t ON g.id = t.entry WHERE t.type = 11");
-    count = 0;
-    if (result)
+        sLog->outString(">> Spawned %u continent motion transports in %u ms", count, GetMSTimeDiffToNow(oldMSTime));
+    }
+    
+    if (sWorld->getBoolConfig(CONFIG_ENABLE_CONTINENT_TRANSPORT_PRELOADING))
     {
-        do
+        // pussywizard: preload grids for continent static transports
+        oldMSTime = getMSTime();
+        result = WorldDatabase.Query("SELECT map, position_x, position_y FROM gameobject g JOIN gameobject_template t ON g.id = t.entry WHERE t.type = 11");
```

Comment:
Here. :)

Reply:
stupid sunwell and their overwriting.. *sigh*